keps/sig-network/784-kube-proxy-component-config/README.md

```diff
@@ -127,6 +131,33 @@ The mitigations to those risks:
 
 ## Design Details
 
+### Changes in structure
+
+| v1alpha1               | v1alpha2                | Comments                                                                    |
+|------------------------|-------------------------|-----------------------------------------------------------------------------|
+|                        | Linux                   | new section for linux (platform-specific) options                           |
+| Conntrack              | Linux.Conntrack         | moved from root to linux (platform-specific) section                        |
+| OOMScoreAdj            | Linux.OOMScoreAdj       | moved from root to linux (platform-specific) section                        |
+| IPTables.MasqueradeBit | Linux.MasqueradeBit     | moved from iptables (backend-specific) to linux (platform-specific) section |
+| NFTables.MasqueradeBit | Linux.MasqueradeBit     | moved from nftables (backend-specific) to linux (platform-specific) section |
+| IPTables.MasqueradeAll | Linux.MasqueradeAll     | moved from iptables (backend-specific) to linux (platform-specific) section |
+| NFTables.MasqueradeAll | Linux.MasqueradeAll     | moved from nftables (backend-specific) to linux (platform-specific) section |
+|                        | Windows                 | new section for windows (platform-specific) options                         | 
+| IPTables.SyncPeriod    | SyncPeriod              | moved from iptables (backend-specific) to root (generic) section            |           
+| NFTables.SyncPeriod    | SyncPeriod              | moved from nftables (backend-specific) to root (generic) section            |
+| IPVS.SyncPeriod        | SyncPeriod              | moved from ipvs (backend-specific) to to root (generic) section             |
+| IPTables.MinSyncPeriod | MinSyncPeriod           | moved from iptables (backend-specific) to root (generic) section            |
+| NFTables.MinSyncPeriod | MinSyncPeriod           | moved from nftables (backend-specific) to root (generic) section            |
+| IPVS.MinSyncPeriod     | MinSyncPeriod           | moved from ipvs (backend-specific) to root (generic) section                |
+| ClusterCIDR            | DetectLocal.ClusterCIDR | moved from root to detect-local section                                     |
+| PortRange              |                         | dropped                                                                     |
+
+### Changes in default values
+
+| Field                       | v1alpha1 (default) | v1alpha2 (default) | 
+|-----------------------------|--------------------|--------------------|
+| IPTables.LocalhostNodePorts | true               | false              |
```

Comment:
`BindAddressHardFail` should go from `false` to `true` too.
And maybe we want to add `ConfigHardFail`, defaulting to `true`, for "kube-proxy exits rather than just warning on config errors".